### DISCUSSION TITLE
unzip: Illegal byte sequence error

### DISCUSSION THREAD

**Discussion by minacle** [Homebrew/discussions#3567] [Writing Formulae/Casks]
I'm writing a cask for font. (gist link: https://gist.github.com/minacle/5722e038318dee5f2d27bec815a9d3cc)

If I run `brew install --debug --cask ./font-gyeonggi-cheonnyeon.rb`, it fails with error:

    ==> Purging files for version latest of Cask font-gyeonggi-cheonnyeon
    Error: Failure while executing; `/usr/bin/env PATH=/opt/homebrew/opt/unzip/bin:/opt/homebrew/Library/Homebrew/shims/shared:/usr/bin:/bin:/usr/sbin:/sbin unzip -qq -o /Users/mayu/Library/Caches/Homebrew/downloads/bb112f84b829fe2aaf632786b7614f933f24656d2f9cf56d1716171d5696ae6c--Gyeonggi_font_OTF_TTF_0602\(Mac\).zip -d /private/tmp/d20220825-29678-epig5x` exited with 50. Here's the output:
    error:  cannot create /private/tmp/d20220825-29678-epig5x/OTF/ۦ�۩���������٬�OTF_Bold.otf
            Illegal byte sequence
    error:  cannot create /private/tmp/d20220825-29678-epig5x/OTF/ۦ�۩���������٬�OTF_Light.otf
            Illegal byte sequence
    error:  cannot create /private/tmp/d20220825-29678-epig5x/OTF/ۦ�۩���������٬�OTF_Medium.otf
            Illegal byte sequence
    error:  cannot create /private/tmp/d20220825-29678-epig5x/OTF/ۦ�۩���������٬�OTFV_Bold.otf
            Illegal byte sequence
    error:  cannot create /private/tmp/d20220825-29678-epig5x/TTF/ۦ�۩���������٬�_Bold.ttf
            Illegal byte sequence
    error:  cannot create /private/tmp/d20220825-29678-epig5x/TTF/ۦ�۩���������٬�_Light.ttf
            Illegal byte sequence
    error:  cannot create /private/tmp/d20220825-29678-epig5x/TTF/ۦ�۩���������٬�_Medium.ttf
            Illegal byte sequence
    error:  cannot create /private/tmp/d20220825-29678-epig5x/TTF/ۦ�۩���������٬�V_Bold.ttf
            Illegal byte sequence
    error:  cannot create /private/tmp/d20220825-29678-epig5x/OTF/ۦ�۩�������٦����OTF_Bold.otf
            Illegal byte sequence
    error:  cannot create /private/tmp/d20220825-29678-epig5x/OTF/ۦ�۩�������٦����OTF_Regular.otf
            Illegal byte sequence
    error:  cannot create /private/tmp/d20220825-29678-epig5x/TTF/ۦ�۩�������٦����_Bold.ttf
            Illegal byte sequence
    error:  cannot create /private/tmp/d20220825-29678-epig5x/TTF/ۦ�۩�������٦����_Regular.ttf
            Illegal byte sequence

How can I resolve this problem?

**Comment by gromgit** [comment]
The problem is with the filenames in your ZIP. They seem to be encoded in some weird scheme that's not UTF-8, so the `unzip` tool bundled with macOS (and most Linux distros too) can't understand them.

For example, the first six characters in your first filename (경기천년제목) should be encoded in UTF-8 as:
```
\xEA\xB2\xBD \xEA\xB8\xB0 \xEC\xB2\x9C \xEB\x85\x84 \xEC\xA0\x9C \xEB\xAA\xA9
```
but it's encoded in your ZIP as:
```
\xDB\xA6\xA2 \xDB\xA9\xA6 \xFD\xA6\xA3 \xD9\xE0\xE4 \xFD\xE1\xA3 \xD9\xAC\xAE
```
You'll have to get help from someone on your side who is experienced with Unicode encodings, specifically with UTF-8. Alternatively, you could package a different ZIP with English-only names, which should cause no issues.

**Reply by minacle** [reply]
추측컨대 UTF-8 대신 CP949로 인코딩 되었을 것입니다.

해당 ZIP 파일은 제가 만든 것이 아니기 때문에 내용물의 파일 이름을 변경하는 것은 어려울 것 같습니다.

`unzip` 도구에서 파일 이름을 해석할 때 어떤 인코딩을 사용할지 지정하는 방법은 없습니까?

**Reply by gromgit** [reply]
Sorry, it seems that the problem is actually with `unzip`. My previous analysis was based on the file listing output by `unzip`, but I just did a raw dump of the ZIP itself, and the Hangul characters are in fact correctly encoded in UTF-8.

Unfortunately, the most straightforward solution is still the same: rename the files with ASCII-only names, then create a new ZIP with the renamed files. Since these are font files, I assume that they will not change very often.
